core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -132,6 +138,43 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
     }
   }
 
+  /** BigQuery data type reference:
+   * <a href="https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types">
+   * Bigquery Standard SQL Data Types</a>
+   */
+  @Override public SqlNode getCastSpec(final RelDataType type) {
+    if (type instanceof BasicSqlType) {
+      SqlUserDefinedTypeNameSpec typeNameSpec;
+      switch (type.getSqlTypeName()) {
+      case BIGINT:
+        return createSqlDataTypeSpecByName("INT64");
+      case DOUBLE:
+        return createSqlDataTypeSpecByName("FLOAT64");
+      case DECIMAL:
+        return createSqlDataTypeSpecByName("NUMERIC");
+      case BOOLEAN:
+        return createSqlDataTypeSpecByName("BOOL");
+      case VARCHAR:
+        return createSqlDataTypeSpecByName("STRING");
+      case VARBINARY:
+        return createSqlDataTypeSpecByName("BYTES");
+      case DATE:
+        return createSqlDataTypeSpecByName("DATE");
+      case TIME:
+        return createSqlDataTypeSpecByName("TIME");
+      case TIMESTAMP:
+        return createSqlDataTypeSpecByName("TIMESTAMP");
+      }
+    }
+    return super.getCastSpec(type);
+  }
+
+  private static SqlDataTypeSpec createSqlDataTypeSpecByName(String identifierName) {
```

Comment:
why static?

Reply:
I usually mark a helper function as a static if it does not access class member. Is there an option on static/non-static in this case?

Reply:
Unless it is a utility class, or the method is called by another static method, I think it should be non-static method. e.g. `Aggregate.isPredicate`

Reply:
Make it as non-static. thanks for the example of Calcite practice.

Reply:
There isn't really a standard practice. Note that a few lines above `Aggregate.isPredicate`, which is static, is `isNotGrandTotal` which is non-static.

In my opinion, if a method is `private` it doesn't really matter. Once it becomes `public` or `protected` it can be overridden, referenced from places that can't see the implementation, etc. So that's when some serious thought has to take place.